OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs

```diff
@@ -251,6 +254,13 @@ public virtual void Activate(Order order)
 			if (power == null)
 				return;
 
+			if (!HasSufficientFunds(power))
+				return;
+
+			var player = Manager.Self;
+			var pr = player.Trait<PlayerResources>();
```

Comment:
Please query and cache this from `Created`, and use `TraitOrDefault` + a null check + remove the `Requires`.